### PR TITLE
feat(services): create section service

### PR DESCRIPTION
**Pull request by aafre**
Create sectionService.ts with 4 pure functions for section operations:
1. getUniqueDefaultName() - generates unique section names (case-insensitive)
2. createDefaultSection() - creates sections with appropriate default content
3. deleteSectionItem() - removes items from section arrays (immutable)
4. reorderSectionItems() - reorders items via drag-drop (uses @dnd-kit/sortable)

Extracted from Editor.tsx:
- Lines 795-822: getUniqueDefaultName logic
- Lines 826-866: createDefaultSection logic
- Lines 693-698: deleteSectionItem logic

Add comprehensive test suite (35 tests):
- getUniqueDefaultName: 8 tests (duplicates, case-insensitive, all types)
- createDefaultSection: 9 tests (all section types + edge cases)
- deleteSectionItem: 8 tests (immutability, edge cases, non-arrays)
- reorderSectionItems: 10 tests (first-to-last, last-to-first, no-op, empty arrays)

Part of Phase 1 - Service Layer for Editor.tsx refactoring

Files:
- New: src/services/sectionService.ts (194 lines, 4 functions)
- New: src/services/__tests__/sectionService.test.ts (445 lines, 35 tests)

Testing:
- Coverage: 99.18% statements, 96.29% branches, 100% functions ✓
- Test suite: 353 tests passed (35 new tests) ✓
- All existing tests still pass ✓

### CODE REVIEW

File: resume-builder-ui/src/services/__tests__/sectionService.test.ts [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+      expect(section.type).toBe('experience');
+      expect(section.name).toBe('New Experience Section');
+      expect(Array.isArray(section.content)).toBe(true);
+      expect(section.content).toHaveLength(1);
+      expect(section.content[0]).toEqual({
+        company: '',
+        title: '',
+        dates: '',
+        description: [''],
+        icon: null,
```
[the remark this answers is not in the document]
Comment (aafre) [reply resume-builder-ui/src/services/__tests__/sectionService.test.ts]: @gemini-code-assist fixed, check now

File: resume-builder-ui/src/services/sectionService.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  let defaultContent;
+  if (type === "experience") {
+    // Default content for Experience sections
+    defaultContent = [
+      {
+        company: "",
+        title: "",
+        dates: "",
+        description: [""],
+        icon: null,
```
[the remark this answers is not in the document]
Comment (aafre) [reply resume-builder-ui/src/services/sectionService.ts]: @gemini-code-assist fixed, check now

File: resume-builder-ui/src/services/__tests__/sectionService.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aafre) [reply resume-builder-ui/src/services/__tests__/sectionService.test.ts]: @gemini-code-assist you sure?
